File: Julia/mlogitOnly/likecalc.jl
function likecalc(Y,X,Z,ßest,N,T,S,J)

N = convert(Int64,N)
T = convert(Int64,T)
S = convert(Int64,S)
J = convert(Int64,J)
P = pclogit(ßest,Y,X,Z,J)

Ywtemp = reshape(Y,N,T,S)
@assert isequal(Ywtemp[:,:,1],Ywtemp[:,:,end])
Pw = zeros(N,T,S,J)
Yw = zeros(N,T,S,J)
for j=1:J
	Pw[:,:,:,j] = reshape(P[:,j],N,T,S,1)
	Yw[:,:,:,j] = Ywtemp.==j;
end

choice_like = ones(N,S)
for s=1:S
	choice_like[:,s] = squeeze(squeeze(prod(prod(Pw[:,:,s,:].^Yw[:,:,s,:],3),2),3),2)
end

full_like = choice_like

return full_like
end

File: Matlab/mlogitOnly/likecalc.m
function full_like = likecalc(Y,X,Z,b,N,T,S,J)

P = pclogit(b,Y,X,Z);

% Yw = permute(reshape(Y,[N S T]),[1 3 2]);
Ywtemp = reshape(Y,[N T S]);
Pw = zeros(N,T,S,J);
Yw = zeros(N,T,S,J);
for j=1:J
	% Pw(:,:,:,j) = permute(reshape(P(:,j),[N S T 1]),[1 3 2 4]);
	Pw(:,:,:,j) = reshape(P(:,j),[N T S 1]);
	Yw(:,:,:,j) = Ywtemp==j;
end

choice_like = ones(N,S);
for s=1:S
	for j=1:J
		choice_like(:,s) = squeeze(prod(prod(Pw(:,:,s,:).^Yw(:,:,s,:),4),2));
	end
end

full_like = choice_like;
end